File: parts/p002/src/test/java/junit/org/rapidpm/event/mutationtesting/p002/ServiceTest.java
package junit.org.rapidpm.event.mutationtesting.p002;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.rapidpm.event.mutationtesting.p002.Service;

/**
 * Create the best test coverage you can for this
 */
class ServiceTest {

    // Cases: a < b, a >= b
    // Case: Loop count <= 0, > 0

  @Test
  void test00NegativeCount() {
    final Service service = new Service();
    Assertions.assertEquals(0, service.tuWas(1,1, -1));
  }

  @Test
  void testZeroCount() {
    final Service service = new Service();
      Assertions.assertEquals(0, service.tuWas(1,1, 0));
  }

    @Test
    void testAsmaller1() {
        final Service service = new Service();
        Assertions.assertEquals(-7, service.tuWas(5,6, 2));
    }

    @Test
    void testAbigger1() {
        final Service service = new Service();
        Assertions.assertEquals(23, service.tuWas(6,5, 2));
    }

}

File: parts/p002/src/test/kotlin/junit/org/rapidpm/event/mutationtesting/p002/ServiceTest.kt
package junit.org.rapidpm.event.mutationtesting.p002

import org.junit.jupiter.api.Test
import org.rapidpm.event.mutationtesting.p002.Service

import org.junit.Assert.assertEquals

/**
 * Create the best test coverage you can for this
 */
internal class ServiceTest {

    @Test
    fun test001() {
        //start here...
        val service = Service()

        val result = service.add(1, 1)

        assertEquals(-2, result.toLong())
    }

    @Test
    fun test002() {
        val service = Service()

        val result = service.add(2, 0)

        assertEquals(2, result.toLong())
    }
}
